### PR TITLE
[MPS] Add nonzero mps support

### PR DESCRIPTION
**Pull request by DenisVieriu97**
Adds nonzero support for mps:

  **Pseudocode**:
  ```
  //
  // inputTensor   = [1,  0,  0,  3]
  // inputNonZero  = [1,  0,  0,  1] (input != 0)
  // scan          = [1,  1,  1,  2] (prefix sum)
  // maskedIndices = [0, -1, -1,  1] (select)
  // coordinates   = [0,  1,  2,  3] (coordinateAlongAxis)
  // scatterResult = [0,  3]         (scatter)
```

### CODE REVIEW

**Review by razarmehr — APPROVED** [review]
LGTM

File: aten/src/ATen/native/mps/operations/Indexing.mm [hunk 1]
```diff
@@ -211,6 +212,185 @@ void index_put_kernel_mps(TensorIterator& iter, IntArrayRef index_size, IntArray
...
+  };
+
+  int64_t total_nonzero = at::count_nonzero(self).item<int64_t>();
+  int64_t nDim = self.dim();
+  at::native::resize_output(out_, {total_nonzero, nDim});
+  if (out_.numel() ==  0) {
+    return out_;
+  }
+
+  bool contiguous_output = (out_.is_contiguous() && !out_.is_view());
```
Comment (malfet) [inline aten/src/ATen/native/mps/operations/Indexing.mm:258]: This feels wrong (i.e. non-contiguous tensors can sometimes be reported as contiguous and views are sometimes contiguous)

**Pull request closed without merging**